janus_server/src/hpke.rs

```diff
@@ -144,38 +144,166 @@ impl HpkeSender {
         plaintext: &[u8],
         associated_data: &[u8],
     ) -> Result<HpkeCiphertext, Error> {
+        let application_info = HpkeApplicationInfo::new(
+            self.task_id,
+            self.label,
+            self.sender_role,
+            self.recipient_role,
+        );
         // We must manually dispatch to each possible specialization of seal
         match (
             self.recipient_config.kem_id,
             self.recipient_config.kdf_id,
             self.recipient_config.aead_id,
         ) {
+            (HpkeKemId::P256HkdfSha256, HpkeKdfId::HkdfSha256, HpkeAeadId::Aes128Gcm) => {
```

Comment:
nit, not caused by this PR: it's a little confusing that the order of the items in the matched tuple is reversed from the order of the items in the generic type parameter list.